celery/__init__.py

```diff
@@ -125,6 +125,16 @@ def _patch_gevent():
         signal.signal = gevent.signal
 
 
+def _patch_threads():
+    try:
+        import tacolib.thread_cancellation
+
+        tacolib.thread_cancellation.patch_all()
+    except (ImportError, AttributeError) as err:
+        sys.stderr.write(f"Cannot patch threads. {err}.\n")
```

Comment:
`_patch_gevent` and `_patch_eventlet` do not catch import errors. It would be bad if for some reason thread worker started without patches and it was not noticed.

Reply:
@DayS1eeper Є варнінг для цього https://github.com/Scalr/fatmouse/blob/spike/celery-threadpool/tacolib/celery/concurrency/thread.py#L183-L187. Для того щоб прибрати обробку потрібно переносити код thread_cancellation в celery.